raiden/network/proxies/payment_channel.py

```diff
@@ -11,22 +12,40 @@ class PaymentChannel:
     def __init__(
         self,
         token_network: TokenNetwork,
-        partner_address: typing.Address,
+        channel_id: typing.ChannelID,
     ):
         self.token_network = token_network
-        self.partner_address = partner_address
+        self.channel_id = channel_id
+
+        # query the blockchain to get the partner addresses for the channel id
+        filter = token_network.proxy.contract.events.ChannelOpened.createFilter(
+            fromBlock=0,
+            argument_filters={
+                'channel_identifier': channel_id,
+            },
+        )
+
+        events = filter.get_all_entries()
+        token_network.proxy.contract.web3.eth.uninstallFilter(filter.filter_id)
+
+        if not len(events) > 0:
+            raise ValueError('Channel is non-existing.')
+
+        event = events[-1]
+        self.participant = decode_hex(event['args']['participant1'])
+        self.partner = decode_hex(event['args']['participant2'])
```

Comment:
for now it's better to maintain the old interface, otherwise this is not a drop-in replacement

Reply:
ignore this, I'm wrong, `partner` has not to do with the old interface.